src/util/metrics/fd_metrics.h

```diff
@@ -0,0 +1,79 @@
+#ifndef HEADER_fd_src_util_metrics_fd_metrics_h
+#define HEADER_fd_src_util_metrics_fd_metrics_h
+
+#include "../fd_util_base.h"
+#include "../../tango/mcache/fd_mcache.h"
+#include <stdbool.h>
+
+struct metrics_t;
+typedef struct metrics_t metrics_t;
+
+typedef enum {
+  metrics_quic,
+  metrics_verify,
+  metrics_dedup,
+  metrics_pack,
+  metrics_bank,
+} metrics_tile_t;
+
+extern char * metrics_tile_names[];
+
+typedef enum {
+  datatype_float,
+  datatype_int,
+  datatype_string,
+  datatype_bool,
+} metrics_datapoint_t;
+
+#define METRICS_VALUE_STRING_SZ 128
+typedef struct {
+  long   i;
+  double f;
+  bool   b;
+  char   s[METRICS_VALUE_STRING_SZ];
+} metrics_value;
+
+#define METRICS_NAME_SZ 40
+
+typedef struct {
+  metrics_datapoint_t type;
+  char                name[METRICS_NAME_SZ];
```

Comment:
Don't think these strings should be passing between tiles, since it's a lot of copying. Could be cleaner to have tag be a u16, then you can eliminate the dcache completely and just send mcache lines which are one cache line. Then metrics tile would map the tags back to strings before posting.